pkg/controller/networkpolicy/adminnetworkpolicy.go

```diff
@@ -0,0 +1,394 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package networkpolicy
+
+import (
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/network-policy-api/apis/v1alpha1"
+
+	"antrea.io/antrea/pkg/apis/controlplane"
+	antreacrd "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	antreatypes "antrea.io/antrea/pkg/controller/types"
+)
+
+var (
+	adminNetworkPolicyTierPriority = int32(251)
+	banpTierPriority               = int32(254)
+	banpPriority                   = float64(1)
+)
+
+func getAdminNPReference(anp *v1alpha1.AdminNetworkPolicy) *controlplane.NetworkPolicyReference {
+	return &controlplane.NetworkPolicyReference{
+		Type: controlplane.AdminNetworkPolicy,
+		Name: anp.Name,
+		UID:  anp.UID,
+	}
+}
+
+func getBANPReference(banp *v1alpha1.BaselineAdminNetworkPolicy) *controlplane.NetworkPolicyReference {
+	return &controlplane.NetworkPolicyReference{
+		Type: controlplane.BaselineAdminNetworkPolicy,
+		Name: banp.Name,
+		UID:  banp.UID,
+	}
+}
+
+// addAdminNP receives AdminNetworkPolicy ADD events and enqueues a reference of
+// the AdminNetworkPolicy to trigger its process.
+func (n *NetworkPolicyController) addAdminNP(obj interface{}) {
+	defer n.heartbeat("addAdminNP")
+	anp := obj.(*v1alpha1.AdminNetworkPolicy)
+	klog.Infof("Processing AdminNetworkPolicy %s ADD event", anp.Name)
```

Comment:
please use structured logging (`klog.InfoS` / `klog.ErrorS`) throughout this PR

Reply:
Done